app/controllers/approved_orders_controller.rb

```diff
@@ -0,0 +1,10 @@
+class ApprovedOrdersController < ApplicationController
+  def create
+    @approved_order = ApprovedOrder.create(params.permit(:order_client))
+    @approved_order.save
+    @order_client = OrderClient.find(params[:order_client_id])
+    @order_client.accepted!
```

Comment:
Essa é uma sugestão mas acho que daria pra melhorar esse código fazendo algo como:

```suggestion
    @order_client = OrderClient.find(params[:order_client_id])
    @approved_order = @order_client.create_approved_order
    @order_client.accepted!
```

Porque aí já cria a approved_order aninhada com a order

Reply:
Feito, valeu!